### ISSUE TITLE
Consider changing __getattr__ "error" return value of ParseResults

### ISSUE THREAD

**Issue by joachimmetz** [pyparsing/pyparsing#208]
From https://github.com/pyparsing/pyparsing/blob/39934574db9796f27397b9f76a112c2466bd1a69/pyparsing/results.py#L359

```
    def __getattr__(self, name):
        try:
            return self[name]
        except KeyError:
            return ""
```

`return ""` causes `TypeError: 'str' object is not callable` to raise in copy.deepcopy() as mentioned here https://github.com/chimpler/pyhocon/issues/220 and 
https://github.com/log2timeline/plaso/issues/2899.

A work-around that I used in the latter issue is to check the return value of `__getattr__` for `if isinstance(value, pyparsing.ParseResults) and not value:` and change the value to None.

**Comment by varunagrawal** [comment]
I have also encountered this issue. One fix that `pyhocon` uses is:

```python
# Fix deepcopy issue with pyparsing
if sys.version_info >= (3, 8):
    def fixed_get_attr(self, item):
        """
        Fix for monkey-patching issue with deepcopy in pyparsing.ParseResults
        """
        if item == '__deepcopy__':
            raise AttributeError(item)
        try:
            return self[item]
        except KeyError:
            return ""

    # apply the monkey-patch
    pyparsing.ParseResults.__getattr__ = fixed_get_attr
```

Which should be the default behavior in `pyparsing`.

**Comment by ptmcg** [comment]
Released in 3.0.0rc2
